streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamTask.java

```diff
@@ -114,18 +115,28 @@ public StreamTask(TaskId id,
         // to corresponding source nodes in the processor topology
         partitionQueues = new HashMap<>();
 
-        TimestampExtractor timestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
+        final TimestampExtractor defaultTimestampExtractor = config.getConfiguredInstance(StreamsConfig.TIMESTAMP_EXTRACTOR_CLASS_CONFIG, TimestampExtractor.class);
 
         for (TopicPartition partition : partitions) {
             SourceNode source = topology.source(partition.topic());
-            RecordQueue queue = createRecordQueue(partition, source, timestampExtractor);
+            // we the source is defined by pattern, we check the sources one by one
```

Comment:
i'd probably extract lines 121 -> 130 into a method, i.e, `findSourceNode(...)`

Also, we -> if